addons/sale_expense/models/hr_expense_sheet.py

```diff
@@ -19,30 +20,36 @@ def _get_sale_order_lines(self):
             :return: sale.order.line
             :rtype: recordset
         """
-        expensed_amls = self.account_move_id.line_ids.filtered(lambda aml: aml.expense_id.sale_order_id and aml.balance >= 0)
+        expensed_amls = self.account_move_id.line_ids.filtered(lambda aml: aml.expense_id.sale_order_id and aml.balance >= 0 and not aml.tax_line_id)
         if not expensed_amls:
             return self.env['sale.order.line']
         aml_to_so_map = expensed_amls._sale_determine_order()
         sale_order_ids = tuple(set(aml_to_so_map[aml.id].id for aml in expensed_amls))
-        aml_sol_unit_price_map = dict(expensed_amls.mapped(lambda aml: (aml.id, aml._sale_get_invoice_price(aml_to_so_map[aml.id]))))
+        aml_sol_unit_price_map = dict(expensed_amls.mapped(
+            lambda aml: (aml.id, formatLang(aml.env, aml._sale_get_invoice_price(aml_to_so_map[aml.id]), aml.currency_id.decimal_places))
+        ))
         product_ids = tuple(expensed_amls.product_id.ids)
         quantities = tuple(expensed_amls.mapped('quantity'))
         names = tuple(expensed_amls.mapped('name'))
         self.env['sale.order.line'].flush_model(['order_id', 'product_id', 'product_uom_qty', 'price_unit', 'name'])
         query = """
             SELECT 
                 DISTINCT ON (sol.order_id, sol.product_id, sol.product_uom_qty, sol.price_unit, sol.name)
```

Comment:
It is generally better to use `GROUP BY` than `DISTINCT ON` because it gives more room to the query planner and doesn't force a `ORDER BY`, especially on big tables like `sale_order_line`